### PR TITLE
Pin ipywidgets

### PR DESCRIPTION
**Pull request by t-imamichi**
### Summary

ipywidgets 8.0.3 (released on Dec 7 [link](https://pypi.org/project/ipywidgets/8.0.3/#history)) raises a deprecation warning as follows.

```
======================================================================
ERROR: test_plot_error_map_over_100_qubit_backend_v2 (test.python.visualization.test_gate_map.TestGateMap)
test.python.visualization.test_gate_map.TestGateMap.test_plot_error_map_over_100_qubit_backend_v2
----------------------------------------------------------------------
testtools.testresult.real._StringException: Traceback (most recent call last):
  File "/Users/ima/tasks/2_2022/qiskit/terra/test/python/visualization/test_gate_map.py", line 151, in test_plot_error_map_over_100_qubit_backend_v2
    fig = plot_error_map(backend)
  File "/Users/ima/tasks/2_2022/qiskit/terra/qiskit/utils/lazy_tester.py", line 165, in out
    return function(*args, **kwargs)
  File "/Users/ima/tasks/2_2022/qiskit/terra/qiskit/utils/lazy_tester.py", line 164, in out
    self.require_now(feature)
  File "/Users/ima/tasks/2_2022/qiskit/terra/qiskit/utils/lazy_tester.py", line 221, in require_now
    if self:
  File "/Users/ima/tasks/2_2022/qiskit/terra/qiskit/utils/lazy_tester.py", line 114, in __bool__
    self._bool = self._is_available()
  File "/Users/ima/tasks/2_2022/qiskit/terra/qiskit/utils/lazy_tester.py", line 289, in _is_available
    imported = importlib.import_module(module)
  File "/usr/local/Cellar/python@3.9/3.9.15/Frameworks/Python.framework/Versions/3.9/lib/python3.9/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 850, in exec_module
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
  File "/Users/ima/envs/dev39/lib/python3.9/site-packages/seaborn/__init__.py", line 12, in <module>
    from .widgets import *  # noqa: F401,F403
  File "/Users/ima/envs/dev39/lib/python3.9/site-packages/seaborn/widgets.py", line 6, in <module>
    from ipywidgets import interact, FloatSlider, IntSlider
  File "/Users/ima/envs/dev39/lib/python3.9/site-packages/ipywidgets/__init__.py", line 25, in <module>
    from .widgets import *
  File "/Users/ima/envs/dev39/lib/python3.9/site-packages/ipywidgets/widgets/__init__.py", line 4, in <module>
    from .widget import Widget, CallbackDispatcher, register, widget_serialization
  File "/Users/ima/envs/dev39/lib/python3.9/site-packages/ipywidgets/widgets/widget.py", line 300, in <module>
    class Widget(LoggingHasTraits):
  File "/Users/ima/envs/dev39/lib/python3.9/site-packages/traitlets/traitlets.py", line 972, in __init__
    cls.setup_class(classdict)
  File "/Users/ima/envs/dev39/lib/python3.9/site-packages/traitlets/traitlets.py", line 1009, in setup_class
    value = getattr(cls, name)
  File "/Users/ima/envs/dev39/lib/python3.9/site-packages/ipywidgets/widgets/widget.py", line 296, in __get__
    return self.fget()
  File "/Users/ima/envs/dev39/lib/python3.9/site-packages/ipywidgets/widgets/widget.py", line 324, in _active_widgets
    deprecation("Widget._active_widgets is deprecated.")
  File "/Users/ima/envs/dev39/lib/python3.9/site-packages/ipywidgets/widgets/utils.py", line 64, in deprecation
    warnings.warn(message, DeprecationWarning, stacklevel=stacklevel+1)
DeprecationWarning: Widget._active_widgets is deprecated.
```

### CODE REVIEW

**Comment by Cryoris** [comment]
I'm a bit confused by this warning: I can reproduce it with running the test but not if I use the function that gives rise to the warning directly:
```python
>>> from qiskit.providers.fake_provider import FakeParis
>>> from qiskit.visualization import plot_error_map
>>> plot_error_map(FakeParis())  # this raises the warning in the test it seems
<Figure size 2400x1800 with 5 Axes>  # no warning here!
```
In the `ipywidgets` it also seems the warning is only raised in a certain [condition](https://github.com/jupyter-widgets/ipywidgets/blob/113f9f36d1d4ca8a4c39273ae5fb045a8cfd01bb/python/ipywidgets/ipywidgets/widgets/widget.py#L323) -- which makes me wonder if the test should be set up differently or whether it's really a issue with ipywidgets 🤔

**Comment by t-imamichi** [comment]
Yes, the condition looks complicated. This PR https://github.com/jupyter-widgets/ipywidgets/pull/3569 introduced it. But I don't figure out it yet.

**Comment by mtreinish** [comment]
Looking at that code it is definitely an issue with ipywidgets (I'm inclined to think it's a bug in their stack inspection code added in that PR) and/or seaborn. The test itself isn't doing anything suspicious in that all that is happening is that seaborn is getting imported as part of the visualization code and triggering a deprecation warning which we treat as fatal in the test suite (so we update our code when an upstream project deprecates an api. If we don't want to pin ipywidgets here the other option is to add the deprecation warnings to the exclude list: https://github.com/Qiskit/qiskit-terra/blob/main/qiskit/test/base.py#L190-L226 which I think is also valid in this case

File: constraints.txt [hunk 1]
```diff
@@ -6,3 +6,7 @@ jsonschema==3.2.0
 # a seaborn import. This pin can be removed when compatibility with those
 # packages is fixed
 jupyter-core==4.11.2
+
+# ipywidgets 8.0.3 started emitting deprecation warnings via a seaborn import.
+# This pin can be removed when compatibility with those packages is fixed.
+ipywidgets<8.0.3
```
Comment (mtreinish) [inline constraints.txt:12]: Was there a case just pinning in the requirements list wasn't enough and we needed to pin it in the constraints file too?

Comment (t-imamichi) [reply constraints.txt:12]: This line upgrades ipywidgets to 8.0.3 without this change of constraints.txt.
https://github.com/Qiskit/qiskit-terra/blob/ee4a5712bfac6104e3e93fc8a3ff845a3a0ddd77/.azure/test-linux.yml#L156

**Comment by t-imamichi** [comment]
Thank you for looking at the code, Matthew. I didn't know about `allow_DeprecationWarning_modules`. You can choose the best way to fix this issue.

File: constraints.txt (hunk 1 above)
Comment (mtreinish) [reply constraints.txt:12]: We probably don't need the requirements-dev.txt change in parallel then. But it doesn't hurt anything either.

**Review by mtreinish — APPROVED** [review]
I'm fine with pinning it's just for testing so that works as well as anything else as long as we remember to circle back after the warnings are fixed

File: constraints.txt (hunk 1 above)
Comment (t-imamichi) [reply constraints.txt:12]: Ah, you are right... I didn't have to change in parallel.

**Pull request merged**